src-tauri/src/commands/item.rs

```diff
@@ -0,0 +1,77 @@
+use rusqlite::params;
+use tauri::{command, State};
+
+use crate::{
+    errors::CommandResult,
+    filters::{get_filtered_items, Filter},
+    schema::Item,
+    AppState,
+};
+
+/// Wraps a basic `SELECT * FROM items` query.
+///
+/// # Examples
+/// ```javascript
+///     let items: Item[] = [];
+///     async function get_items() {
+///         items = await invoke("get_items");
+///     };
+/// ```
+///
+#[command]
+#[allow(clippy::needless_pass_by_value)]
+pub fn get_items(state: State<AppState>, filter: Option<Filter>) -> CommandResult<Vec<Item>> {
+    let db = state.db.conn.lock().unwrap();
+    let mut items = Vec::new();
+    if let Some(filter) = filter {
+        items = get_filtered_items(&db, filter).unwrap();
+    } else {
+        println!("No Filter!");
+        let mut stmt = db.prepare("SELECT * FROM items")?;
+        let rows = stmt.query_map([], |row| {
+            Ok(Item {
+                id: row.get(0)?,
+                name: row.get(1)?,
+                description: row.get(2).unwrap_or(String::new()),
+                comments: row.get(3).unwrap_or(String::new()),
+                date: row.get(4)?,
```

Comment:
Nice use of `.get`

Are we handling the `None` value somewhere?

Reply:
Currently any unhandled errors/nones get passed up to the frontend as a Serialized string and are printed in the console. id, name, and date are all NOT NULL fields and if they are NULL then we might even want to panic since the database is severely out of spec with its own rules. 

Theres also a [crate](https://docs.rs/serde_rusqlite/latest/serde_rusqlite/) that will serialize a sqlite row into a struct for you, that could be a great option to hand all this error checking off to a separate crate.